codice/ddf
# DDF-3632 clean up resource cache and set storage directory in blueprint

#### What does this PR do?

* Fixes setting product cache directory
* Cleans up tests that were writing to the target directory
* Cleans up usages of system properties in tests and production code
* Stops resource cache from relying on the bundle context in order to get hazelcast configuration

#### Who is reviewing it? 
@peterhuffer @clockard @kcover 

#### Choose 2 committers to review/merge the PR. 
@figliold @lessarderic 

#### How should this be tested? (List steps with links to updated documentation)
Test default directory:
1. Start the system, install standard profile.
2. Configure a loopback csw source
3. Upload a file through Intrigue
4. Query loopback source
5. download product
6. check contents of `$DDF_HOME/data/Product_Cache`

Test alternate directory:
1. In admin console, go to the catalog-app configurations, click `Resource Download Settings`
2. Set `Product Cache Directory` to something other than the default
3. upload new file in intrigue
4. query and retrieve from loopback source
5. verify contents of new cache directory

#### What are the relevant tickets?
[DDF-3632](https://codice.atlassian.net/browse/DDF-3632)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
